scripts/deployment/_deploy-lock-request

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+. ./scripts/devcontainer/_assert-in-container "$0" "$@"
+
+# Add a lock identifier for the commit and the timestamp.
+
+set -euo pipefail
+
+if [ ! -v DEPLOY_LOCK_TOKEN ] ;
+then
+  echo "No deploy lock token, get it from https://darklang.com/a/ops-circleci"
+  exit 1
+fi
+
+REQUEST_PATH=$1
+METHOD=$2
+shift 2
```

Comment:
I haven't used `shift` before. Could you explain what this is doing? I researched a bit, but confused - knowing what it does in this specific case would help.

Reply:
It "shifts" 2 items (pops from the front of the list). The list is implicit, it's `$@`. So this removes $1 and $2 from $@ so that we can use $@ later.

Reply:
I simplified the code and removed the shift

Reply:
I appreciate it - somehow `shift` feels like magic to me a bit :)